### ISSUE TITLE
improving biases when browsing internet for images

### ISSUE THREAD

**Issue by michaeljoelt** [borisdayma/dalle-mini#235]
I searched "doctor" and it returned the following:
![Screenshot_20220612-104224](https://user-images.githubusercontent.com/21367319/173246154-26f1ca08-f832-4583-8486-3457e53b5cc6.png)

for when general human images are being searched, it seems it should attempt to include more diversity. so instead of only looking up "doctor", it should also look up: "black doctor", "brown doctor", "asian doctor", "woman doctor", etc.

**Comment by IronMaltese** [comment]
If the site is changing anything about the input, that should be made clear to the user

**Comment by redelman431** [comment]
Right now there is already complaints from autism rights groups about some of the results that show up in the search engine. I would strongly advise monitoring the social media and blogsphere for any further complaints about bias. Perhaps we should make this a permanently open issue so we can continue to address those problems and nip them in the bud. Or perhaps create a separate communication channel on the website to deal with those specific types of complaints.

![Uploading inbound4315208440336286872.jpg…]()

**Comment by michaeljoelt** [comment]
Yes, disability related biases are another great point. I have actually been noticing quite a few folks generating immature and hateful images on the discord bot.. often related to Nazis, making fun of LGBT, disabilities, etc... not sure what the route is for reporting that, if there's one? interesting how obsessed they are with these topics though... I guess. 

and yeah, regarding the other comment: "weighted" searches are necessary when the "default" searches are super biased. if you want a "white doctor" specifically, you can request it.

**Comment by philiparvidsson** [comment]
I don't think it's feasible to address this in the scope of generating images using AI.

- If 90% of doctors in real-life society are white, and the model generates white doctors, is that a bias?
- If 90% of doctors in real-life society are white, is that in and of itself a bias?
- If 90% of doctors in real-life society are white and the model is generating 50% doctors with a different skin color, haven't we actually *introduced* a bias in the model? How do we know if it is more or less biased if it does so, given the only fact you have is that 90% doctors are white?

**How can we tell what is a bias, what isn't a bias, what is a correct counter-bias and what is an incorrect counter-bias?**

This is a deeply psychological issue and it cannot be addressed properly by telling the model to generate at least 50% female colored doctors - it does not solve the issue of the claim that the model has a bias because it is generating white male doctors. Here are a few things to think about:

1. The claim that generating white male doctors is a bias when real-life society consists of 90% white male doctors is false. There might be a bias, but the bias is not in copying reality - it is deep within the structures of society itself, assuming it is even there to begin with (how would we even know!?).

2. There is no way to verify that introducing a counter-bias to generate 50% colored female doctors is much less biased than 90% white male doctors. What if, without our psychological biases, we would have 100% colored female doctors? What makes you think 50% is a good amount of diversity other than societal propaganda of the times we are living in? Maybe, in the future, ALL doctors will be colored. Why should it be allowed to generate any white doctors?

3. Why should the model generate anything other than what it can observe? That gives MORE power to AI to control the direction society is going in, not less. 50% of doctors are not colored females. Why do you want it to generate that? To make a point that colored females can be doctors too? That's called propaganda or brain washing. You are telling the AI to generate something that _you_ think it should generate, _to plant thoughts in the heads of whoever is using the AI_. How is that any less evil than being biased or even racist?

4. Who decides that it should be 50% colored doctors? Why can't we tell it to only generate white male doctors wearing swastiskas? Is that more or less truthful than generating 50% colored female doctors? How can we tell, or measure it?

5. Is it important for the model to be truthful? If not, then we are in essence building a propaganda tool. That gives enormous political power to anyone deciding over what the AI gets to generate. Why do you want the AI to wield that kind of power?

6. What if we told it to only generate images of a flat earth when generating planets? If that was the consensus political opinion, should the model be told to only generate flat earth imagery? If science says the planet is a globe, should that rule over political opinion? What if science says that there are difference between different races? Should the AI be allowed to reflect that or not?

Those are just some questions that I came up with re. this topic. It's a really difficult topic and it needs many thousands of hours of rumination by our brightest minds, not political opinions by the masses.

TL;DR: You have no f****** idea of what you are getting yourself into.

DISCLAIMER: I do not know about doctor demographics and the 90% white male doctor example is only to make a point, it is not what I believe the doctor demographic to be.

**Comment by janicetyler** [comment]
> Why should the model generate anything other than what it can observe? like this https://github.com/patsybond172 or this https://patsybond172.github.io/kitchen-cabinets

i think i same with you

**Comment by janicetyler** [comment]
yeah, regarding the other comment: "weighted" searches are necessary when the "default" searches are super biased. if you want a "white doctor" specifically, you can request it.
[https://github.com/janicetyler](https://github.com/janicetyler)
[https://github.com/janicetyler/wood-kitchen-cabinets](https://github.com/janicetyler/wood-kitchen-cabinets)
[https://janicetyler.github.io/wood-kitchen-cabinets](https://janicetyler.github.io/wood-kitchen-cabinets/)

**Comment by lucialam172** [comment]
Why should the model generate anything other than what it can observe? like this 
[https://github.com/lucialam172/dark-kitchen-cabinets](https://github.com/lucialam172/dark-kitchen-cabinets)
[https://github.com/lucialam172](https://github.com/lucialam172)
[https://lucialam172.github.io/dark-kitchen-cabinets](https://lucialam172.github.io/dark-kitchen-cabinets)
